### ISSUE TITLE
test units / suite?

### ISSUE THREAD

**Issue by cusspvz** [bendc/sprint#10]
I've issued because stability and performance are my top criteria.

**Comment by haoxins** [comment]
+1

**Comment by cusspvz** [comment]
PS: I've only issued because stability and performance are my top criteria.

**Comment by Charca** [comment]
+1

**Comment by stephnr** [comment]
I am willing to help with writing unit tests. Is there a consensus for framework of choice?

I would prefer either of the following:
1. [Mocha](http://mochajs.org/)
2. [Jasmine](http://jasmine.github.io/)

All wrapped in [Karma](http://karma-runner.github.io/0.12/index.html)

**Comment by cromulus** [comment]
:+1:

**Comment by kitten** [comment]
As I've stated in #31, we could try to utilise [Jest](https://github.com/facebook/jest). It's based on Jasmine, but like its website says very painless to use.

**Comment by stephnr** [comment]
Please see #31 for the new branch using the Jest testing framework as proposed by @philplckthun. Thank you for the advice. This was exactly what I needed. Jasmine alone did not provide enough functionality for CLI support. I look forward to seeing everyones contributions :+1:

**Comment by gmac** [comment]
Yep, another :+1: on this. Its a glaring omission when a tool in consideration doesn't ship with a robust test suite.

**Comment by stephnr** [comment]
For those who do wish to contribute, the Testing branch can be located here:

https://github.com/Stephn-R/sprint
